File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/CiaCommandConstants.java
package top.ourisland.creepersiarena.core.command;

import java.util.List;

/**
 * Shared command literals, aliases and small static suggestion sets.
 *
 * <p>The first command refactor stage keeps the runtime behaviour intact while
 * moving previously inline constants out of the command tree implementation.</p>
 */
public final class CiaCommandConstants {

    public static final String PLAYER_ROOT_LITERAL = "creepersia";
    public static final String ADMIN_ROOT_LITERAL = "ciaa";
    public static final String ADMIN_EMBEDDED_LITERAL = "admin";

    public static final List<String> PLAYER_ROOT_ALIASES = List.of("cia", "creepersiarena");
    public static final List<String> NO_ALIASES = List.of();
    public static final List<String> TEAM_SUGGESTIONS = List.of(
            "random",
            "1", "2", "3", "4",
            "red", "blue", "green", "yellow",
            "aqua", "cyan", "purple", "white", "black"
    );
    public static final List<String> PLAYER_LANGUAGE_SUGGESTIONS = List.of("default", "en_us", "zh_cn");
    public static final List<String> ADMIN_LANGUAGE_SUGGESTIONS = List.of("en_us", "zh_cn");
    public static final List<String> CONFIG_TARGET_SUGGESTIONS = List.of("config", "arena", "skill");
    public static final List<String> BOOLEAN_SUGGESTIONS = List.of("true", "false");

    private CiaCommandConstants() {
    }

}

File: cia-core/src/main/java/top/ourisland/creepersiarena/core/command/CiaCommandConstants.kt
package top.ourisland.creepersiarena.core.command

/**
 * Shared command literals and small static suggestion sets.
 */
object CiaCommandConstants {

    const val PLAYER_ROOT_LITERAL: String = "cia"
    const val ADMIN_ROOT_LITERAL: String = "ciaa"

    @JvmField
    val PLAYER_ROOT_ALIASES: List<String> = listOf()

    @JvmField
    val ADMIN_ROOT_ALIASES: List<String> = listOf()

    @JvmField
    val PLAYER_LANGUAGE_SUGGESTIONS: List<String> = listOf("default", "en_us", "zh_cn")

    @JvmField
    val ADMIN_LANGUAGE_SUGGESTIONS: List<String> = listOf("en_us", "zh_cn")

    @JvmField
    val CONFIG_TARGET_SUGGESTIONS: List<String> = listOf("config", "arena", "skill")

    @JvmField
    val BOOLEAN_SUGGESTIONS: List<String> = listOf("true", "false")

    @JvmField
    val PREFERENCE_BOOLEAN_SUGGESTIONS: List<String> = listOf("on", "off", "true", "false")

    @JvmField
    val TEAM_SUGGESTIONS: List<String> = listOf(
        "random",
        "1", "2", "3", "4",
        "red", "blue", "green", "yellow",
        "aqua", "cyan", "purple", "white", "black"
    )

}
